packages/data-explorer/src/store/actions.ts

```diff
@@ -26,45 +26,52 @@ export default {
       await dispatch('header/fetchBreadcrumbs')
     }
   },
-  async fetchViewData ({ dispatch, state }: any) {
-    if (state.dataDisplayLayout === 'CardView') {
-      dispatch('fetchCardViewData')
-    } else {
-      dispatch('fetchTableViewData')
-    }
-  },
-  fetchCardViewData: async ({ commit, state, getters }: { commit: any, state: ApplicationState, getters: any }) => {
-    commit('setLoading', true)
+  async fetchViewData ({ state, dispatch, commit }) {
     if (state.tableName === null) {
-      commit('addToast', { message: 'cannot load card data without table name', type: 'danger' })
-      commit('setLoading', false)
+      commit('addToast', { message: 'cannot fetch table view data without table name', type: 'danger' })
       return
     }
 
     if (state.tableMeta === null) {
-      commit('addToast', { message: 'cannot load card data without meta data', type: 'danger' })
-      commit('setLoading', false)
+      commit('addToast', { message: 'cannot fetch table view data without meta data', type: 'danger' })
       return
     }
+    commit('setLoading', true)
+
+    if (state.dataDisplayLayout === 'CardView') {
+      dispatch('fetchCardViewData')
+    } else {
+      dispatch('fetchTableViewData')
+    }
+  },
+  /**
+   * Don't call this function directly (except for tests), use fetchViewData instead
+   */
+  fetchCardViewData: async ({ commit, state, getters }) => {
+    const { tableName, tableMeta, tablePagination, sort, tableSettings, dataDisplayLayout } = state
```

Comment:
destructuring leads to vars being declared to soon, out side of context and possibly unused

Reply:
Destructuring gives us more clarity upfront and removes unnecessary trailing paths, it's aready happening in the function.
unnecessary vars isn't an issue, as you only export what you need. And if there would be a rogue variable, the IDE will tell you so.